src/s32-vmultiply/s32-vmultiply.c.in

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 Google LLC
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+$ABC = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+$BATCH_TILES = tuple(int(bt) for bt in BATCH_TILES.split(","))
+$SIMD_SIZE = BATCH_TILES[0]
+#include <assert.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "xnnpack/simd/s32-${ARCH}.h"
+
+#include "xnnpack/common.h"
+#include "xnnpack/microparams.h"
+#include "xnnpack/vunary.h"
+
+$for BATCH_TILE in BATCH_TILES:
+  $assert BATCH_TILE % SIMD_SIZE == 0
+  $assert BATCH_TILE >= SIMD_SIZE
+  $SIMD_TILE = BATCH_TILE // SIMD_SIZE
+
+  void xnn_s32_vmultiply_ukernel__${ARCH}_u${BATCH_TILE}(
+      size_t batch,
+      const int32_t* input_a,
+      const int32_t* input_b,
+      int32_t* output,
+      const union xnn_s32_default_params unused_params[restrict XNN_MIN_ELEMENTS(1)])
+  {
+    assert(batch != 0);
+    assert(batch % sizeof(int32_t) == 0);
+    assert(input_b != NULL);
+    assert(input_a != NULL);
+    assert(output != NULL);
+    assert(xnn_simd_size_s32 == ${SIMD_SIZE});
+
+
+    $if SIMD_TILE > 1:
+      for (; batch >= ${BATCH_TILE} * sizeof(int32_t); batch -= ${BATCH_TILE} * sizeof(int32_t)) {
+        xnn_simd_s32_t vin1_${ABC[0]} = xnn_loadu_s32(input_a);
+        $for N in range(1, SIMD_TILE):
+          xnn_simd_s32_t vin1_${ABC[N]} = xnn_loadu_s32(input_a + ${N} * xnn_simd_size_s32);
+        input_a += ${BATCH_TILE};
+
+        xnn_simd_s32_t vin2_${ABC[0]} = xnn_loadu_s32(input_b);
+        $for N in range(1, SIMD_TILE):
+          xnn_simd_s32_t vin2_${ABC[N]} = (xnn_loadu_s32(input_b + ${N} * xnn_simd_size_s32));
+        input_b += ${BATCH_TILE};
+
+        $for N in range(0, SIMD_TILE):
+          xnn_simd_s32_t vy_${ABC[N]} = xnn_mul_s32(vin1_${ABC[N]}, vin2_${ABC[N]});
```

Comment:
Clamping must be handled here like in other vmul microkernels

Reply:
Added a clamping wrapper files